spec/features/transfer_account_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'rails_helper'
+
+RSpec.describe 'transfer user account ownership', type: :feature do
+  let(:user) { create(:user, :with_jwt) }
+  let(:form_dom_id) { "#transfer_account_user_#{user.id}" }
+  let(:email) { 'valid@email.com' }
+
+  subject do
+    visit user_profile_path
+    within form_dom_id do
+      fill_in 'email', with: email
+      click_button 'Confirmer'
+    end
+  end
+
+  before { log_as(user) }
+
+  it 'calls the underlying operation' do
```

Comment:
Je suis un peu vieux jeu, mais sur du test de feature j'aime bien tester fonctionnellement en end-to-end, et du coup vérifier qu'il y a bien eu un transfert effectif.

Après, c'est juste pour m'assurer que l'opération est un succès (je test pas tous les aspects de l'opération), en mockant c'est pas impossible que ça ne fasse pas ça.

Reply:
Je sais pas trop encore quoi tester dans les features, j'avais simplement adapté les tests des contrôleurs API existants. L'opération elle même est déjà bien testée et je voulais éviter les doublons. 

<img width="1440" alt="Capture d’écran 2021-10-05 à 12 22 39" src="https://user-images.githubusercontent.com/4283958/136005594-fe1eadc2-e541-4d3b-b45b-bb84ce3c8b21.png">

Je veux bien ton avis sur les tests sus jacent que tu verrais bien en tests de feature.

Reply:
C'est vraiment lié à l'expérience que j'ai sur les tests de features: en mockant le service dans le controller je me suis déjà retrouvé à avoir des tests de features green alors que ça ne fonctionnait pas, parce que mes tests n'étaient pas assez robuste.

Typiquement ici on s'assure du succès à travers le test d'une alerte `notice` ( ici https://github.com/etalab/admin_api_entreprise/pull/189/files/3f7e7070347a25556dfc3c4db126571de1e74501#diff-6bb650c47e2e6aaf1f991990ce359884361fa0e405a7f50a2d92e9cbfe105a58R48 ). Plus tard on peut se retrouver à avoir du faux négatif.

Donc franchement sur ce genre de situation, je n'ai pas d'avis tranché et je laisse à la discrétion du dev le choix de tester un morceau fonctionnel ou non. Dans ce cas là, je ferai le truc le plus obvious, qui est de vérifier qu'un jeton n'appartient plus à l'utilisateur courant.